src/test/java/se/sundsvall/comfactfacade/service/SigningServiceTest.java

```diff
@@ -1,56 +1,129 @@
 package se.sundsvall.comfactfacade.service;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.nio.charset.StandardCharsets;
 
 import org.junit.jupiter.api.Test;
 import org.junit.jupiter.api.extension.ExtendWith;
 import org.mockito.InjectMocks;
+import org.mockito.Mock;
 import org.mockito.junit.jupiter.MockitoExtension;
 
-import se.sundsvall.comfactfacade.TestUtil;
+import se.sundsvall.comfactfacade.api.model.SigningRequest;
+import se.sundsvall.comfactfacade.integration.comfact.ComfactIntegration;
+
+import comfact.DocumentType;
+import comfact.GetSignatoryRequest;
+import comfact.SigningInstance;
 
 @ExtendWith(MockitoExtension.class)
 class SigningServiceTest {
 
+	@Mock
+	private ComfactIntegration comfactClient;
+
 	@InjectMocks
 	private SigningService signingService;
 
 	@Test
 	void createSigningRequest() {
+		// Arrange
+		when(comfactClient.createSigningInstance(any())).thenReturn(new comfact.CreateSigningInstanceResponse().withSigningInstanceId("123"));
 
-		final var result = signingService.createSigningRequest(new TestUtil.SigningRequestBuilder().build());
-		assertThat(result).isNotNull().isEqualTo("OK");
+		// Act
+		final var result = signingService.createSigningRequest(new SigningRequest());
+
+		// Assert
+		assertThat(result).isNotNull().isEqualTo("123");
 	}
 
 	@Test
-	void cancelSigningRequest() {
+	void updateSigningRequest() {
+		// Arrange
+		final var signingId = "someSigningId";
+		final var signingRequest = new SigningRequest();
+
+		// Act
+		signingService.updateSigningRequest(signingId, signingRequest);
+
+		// Assert
+		verify(comfactClient).updateSigningInstance(any());
```

Comment:
Use captor to verify that correct values are propagated to client mock?

Comment goes for all calls to client.